students/jeff_shabani/session08/circle.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import math
+
+"""
+Framework for a circular object.
+Validation prevents user from entering a 
+negative radius value.
+"""
+
+
+class Circle(object):
+    instances = []
+
+    def __init__(self, radius):
+        if radius < 0:
+            raise ValueError('It is not possible for a radius to be less than zero.')
+        else:
+            self.radius = radius
+        self._radius = radius
+        Circle.instances.append(self)
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    # make area non-settable
+    @property
+    def area(self):
+        return math.pi * pow(self.radius, 2)
+
+    # method to create a circle with the diameter
+    @classmethod
+    def from_diameter(cls, value):
+        radius = value / 2
+        return cls(radius)
+
+    # simple add method
+    def __add__(self, other):
```

Comment:
Hi Jeff,

These arithmetic methods are expected to return new Circle objects, not integers. In this case, adding a Circle(2) object to a Circle(4) object should return a Circle(6) object (a new object whose radius is the sum of those of the original circles). A similar logic goes for _mul_.

This will also require updating your tests to reflect that the expected results are also Circles.

Regards,

Luis